table/table_factory.cc

```diff
@@ -37,6 +37,7 @@ Status TableFactory::CreateFromString(const ConfigOptions& config_options_in,
     factory->reset(new CuckooTableFactory());
 #endif  // ROCKSDB_LITE
   } else {
+    status.PermitUncheckedError();
     return Status::NotSupported("Could not load table factory: ", name);
   }
   if (!existing_opts.empty()) {
```

Comment:
Why not status = Status::NotSupported()
and on line 42/3 if (status.ok() && !existing_opts.empty())